Source/WebCore/page/Navigation.cpp

```diff
@@ -396,6 +456,35 @@ static bool documentCanHaveURLRewritten(const RefPtr<Document>& document, const
     return equalIgnoringFragmentIdentifier(documentURL, targetURL);
 }
 
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#promote-an-upcoming-api-method-tracker-to-ongoing
+void Navigation::promoteUpcomingAPIMethodTracker(const String& destinationKey)
+{
+    ASSERT(!m_ongoingAPIMethodTracker);
+
+    if (!destinationKey.isNull()) {
+        // FIXME: ASSERT(!m_upcomingNonTraverseMethodTracker);
+        m_ongoingAPIMethodTracker = m_upcomingTraverseMethodTrackers.getOptional(destinationKey);
+        m_upcomingTraverseMethodTrackers.remove(destinationKey);
+    } else {
+        m_ongoingAPIMethodTracker = m_upcomingNonTraverseMethodTracker;
+        m_upcomingNonTraverseMethodTracker = std::nullopt;
+    }
+}
+
+// https://html.spec.whatwg.org/multipage/nav-history-apis.html#navigation-api-method-tracker-clean-up
+void Navigation::cleanupAPIMethodTracker(NavigationAPIMethodTracker& apiMethodTracker)
```

Comment:
Is apiMethodTracker actually being changed?

Reply:
It is being removed from the Navigation object, so gets destroyed.